lib/web_ui/test/canvaskit/initialization_services_vs_ui_test.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html' as html;
+
+import 'package:test/bootstrap/browser.dart';
+import 'package:test/test.dart';
+import 'package:ui/src/engine.dart';
+
+import 'common.dart';
+
+void main() {
+  internalBootstrapBrowserTest(() => testMain);
+}
+
+void testMain() {
+  test('services are initalized separately from UI', () async {
+    expect(scheduleFrameCallback, isNull);
+    expect(windowFlutterCanvasKit, isNull);
+
+    expect(findGlassPane(), isNull);
+    expect(Keyboard.instance, isNull);
+    expect(MouseCursor.instance, isNull);
+    expect(KeyboardBinding.instance, isNull);
+    expect(PointerBinding.instance, isNull);
+
+    // After initializing services the UI should remain intact.
+    await initializeEngineServices();
+    expect(scheduleFrameCallback, isNotNull);
```

Comment:
I wish there was a way to compute the CRC of a node (or similar) to ensure things haven't changed in a DOM tree before/after calling this method :/

Reply:
That's an interesting idea. We do have utilities for comparing HTML. If we add shadow DOM support to it we could use it to compare before/after. I wouldn't include that in this PR though.